modules/distributing/slides-distributing.md

```diff
@@ -15,96 +15,465 @@ author: Jaro Camphuijsen, Lourens Veen
+even EasyBuild, Spack or Nix (on High-Performance Computing)
+
+===
+
+<!-- .slide: data-state="standard" -->
+
+### Scenario 3: A whole application
+
+- Not so common in science
+- You have created a program
+- To be used by end users
+
+note:
+Scientists don't often make whole applications with user interfaces that anyone can just
+pick up and use. That's a lot of work, and you don't get credit for it in science.
+Nevertheless there are examples of this, e.g. MaxEnt in ecology, and it could happen. In
+this case, what you have is a program.
+
+===
+
+<!-- .slide: data-state="standard" -->
+
+### Distributing programs
+
+- In an app store!
+- As an installer?
+- As a package?
+- As an image?
+
+note:
+There are a few options for distributing a program. Packaging systems can be used,
+especially if the program is intended for use by programmers, or by more knowledgeable
+users. An App Store is basically a packaging system for applications, and that would be
+a good option too if available.
+
+Another option is to make an installer. That's a single file that is a working program,
+like a package, but it is also itself a program that when run will install your program
+onto the user's system. If you've installed software on Windows then you may have seen
+one, although they've mostly been replaced by app stores and my cause the operating
+system to warn about suspicious software.
+
+Finally, you could make an image here, but as before, it would be very large, and it
+would require special software on the user's computer, so in the here and now it's not a
+good option. It would be more resistant to changes in the operating system that break
+the program, because it includes the old version of the operating system. And then
+again, running an old operating system is a security risk.
+
+===
+
+<!-- .slide: data-state="standard" -->
+
+## Updating software
+
+- Everything changes
+- For scripts, you probably can
+- Libraries and programs need maintenance
 
-- requirements.txt
-- environment.yml
-- pyproject.toml
-- package.json
-etc...
 
 note:
-Specify the dependencies in a file and let the user build their own environment, container or vm.
+Scripts are built on top of libraries, which sit on top of an operating system, which
+sits on top of hardware, and all that stuff is constantly changing in the name of
+progress. Furthermore, people change and expect to use software in different ways over
+time, and that also needs to be taken into account.
+
+For example, the LimePy library in the AMUSE astrophysics software uses a function in
+the SciPy library that does not exist anymore. The old version of SciPy it was written
+for was designed for an old version of Python, which is no longer supported and
+shouldn't be used anymore, for security reasons. Also, other software my script uses may
+require the newer version, and you can't install multiple versions of the same thing at
+the same time. So that script doesn't work anymore.
```

Comment:
I think either call it "my script" throughout, or call it by its name throughout.

Reply:
These are different things, 'my script' uses LimePy. I think it may help to clarify that the versions mentioned are of SciPy however.

Reply:
Reworded, should be better now.